### PR TITLE
Add guide for check output metric extraction

### LINKED ISSUE #1388: [Metric Extraction] Documentation
Child of https://github.com/sensu/sensu-go/issues/745
Depends on #1384 

Write a guide on how to extract metrics from check output in [sensu/sensu-docs](https://github.com/sensu/sensu-docs). The guide should cover some basics on the supported line protocols and formats, and how to configure a check that you wish to extract metrics from.

### PR DESCRIPTION
**Pull request by nikkictl**
Signed-off-by: Nikki Attea <nikki@sensu.io>

Creates a step by step guide for extracting metrics from check output.

## Description
Intro to check and metrics, supported metric formats, and instructions for metric extraction.

## Motivation and Context
Closes https://github.com/sensu/sensu-go/issues/1388.

## How Has This Been Tested?
CI.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New doc/guide
- [ ] Fixing errata

## Checklist:
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] All tests have passed.

### CODE REVIEW

File: content/sensu-core/2.0/guides/extract-metrics-with-checks.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+sensuctl check set-metric-format collect-metrics graphite_plaintext
+sensuctl check set-metric-handlers collect-metrics influx-db
+{{< /highlight >}}
+
+### Supported metric formats
+
+The metric formats that Sensu currently supports for check output metric
+extraction are nagios, influxdb, graphite, and opentsdb.
+
+nagios           |
```
Comment (palourde) [inline content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: Looks like there's a formatting issue here!
screen shot 2018-05-07 at 10 49 27 am

File: content/sensu-core/2.0/guides/extract-metrics-with-checks.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+If the check output is formatted correctly according to its `metric_format`,
+the metrics will be extracted in Sensu Metric Format, and saved within the
+event. You should expect to see logged errors if Sensu is unable to parse
+the check output. You can validate that metrics have been extracted from your
+check through your handler, or through the resulting event. The example check
+we used would yield an event similar to the one below:
+
+{{< highlight json >}}
+"event": {
```
Comment (palourde) [inline content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: We need an opening bracket (`{`) here and a closing one at the end of this block!

File: content/sensu-core/2.0/guides/extract-metrics-with-checks.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  "check": {
+    "name": "collect-metrics",
+    "command": "collect_metrics.sh",
+    "output": "cpu.idle_percentage 61 1525462242\nmem.sys 104448 1525462242
```
Comment (palourde) [inline content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: There's a missing quote & comma at the end of this line!

**Review by palourde — CHANGES_REQUESTED** [review]
Just some small concerns around formatting, otherwise the content is 👍

File: content/sensu-core/2.0/guides/extract-metrics-with-checks.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+### Supported metric formats
+
+The metric formats that Sensu currently supports for check output metric
+extraction are nagios, influxdb, graphite, and opentsdb.
+
+|nagios       |      |
+--------------|------
+metric_format | nagios_perfdata
+documentation | [Nagios Performance Data][6]
+example       | `PING ok - Packet loss = 0%, RTA = 0.80 ms | percent_packet_loss=0, rta=0.80`
```
Comment (palourde) [inline content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: You'll probably need to somehow escape the `|` before `percent_packet_loss`

Comment (nikkictl) [reply content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: Yeah I _just_ noticed that and I'm trying to figure out how haha

Comment (palourde) [reply content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: @nikkiki You could do something like this I think:

```
example       | {{< highlight bash >}}PING ok - Packet loss = 0%, RTA = 0.80 ms | percent_packet_loss=0, rta=0.80{{< /highlight >}}
```

Comment (nikkictl) [reply content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: Yeah I chose to do `string` over `bash` since highlighting isn't totally necessary.

Comment (portertech) [reply content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: String is probably best, as bash could cause some funk.

Comment (palourde) [reply content/sensu-core/2.0/guides/extract-metrics-with-checks.md]: Oh yeah totally good point!

**Review by palourde — APPROVED** [review]
